backend/vxlan/vxlan_network_windows.go

```diff
@@ -87,36 +95,66 @@ func (nw *network) handleSubnetEvents(batch []subnet.Event) {
 			continue
 		}
 
-		publicIP := leaseAttrs.PublicIP.String()
-		remoteIP := leaseSubnet.IP + 2
-		lastIP := leaseSubnet.Next().IP - 1
+		var vxlanAttrs vxlanLeaseAttrs
+		if err := json.Unmarshal(leaseAttrs.BackendData, &vxlanAttrs); err != nil {
+			log.Error("error decoding subnet lease JSON: ", err)
+			continue
+		}
+
+		hnsnetwork, err := hcn.GetNetworkByName(nw.dev.link.Name)
+		if err != nil {
+			log.Errorf("Unable to find network %v, error: %v", nw.dev.link.Name, err)
+			continue
+		}
+		managementIp := event.Lease.Attrs.PublicIP.String()
+
+		networkPolicySettings := hcn.RemoteSubnetRoutePolicySetting{
+			IsolationId:                 4096,
+			DistributedRouterMacAddress: net.HardwareAddr(vxlanAttrs.VtepMAC).String(),
```

Comment:
Just a question. How was it even working with ConjureMAC before? Not ARP right? Or was the MAC manufactured and dropped on the interface?

Reply:
Previously there was a remote endpoint for each IP in the subnet on the host. For example if Host A has subnet 192168.1.0/24 and Host B has 192.168.2.0/24, then Host A had 250+ remote endpoints from 192.168.2.1-254. Vice versa for Host B

This was before HNS had the remote subnet feature. And before Remote Subnet, many things were not working properly like service vip, node port access, etc. With remote subnet feature, this RemoteSubnetPolicy replaces the tens/hundreds of endpoints you previously had to create per other node in the cluster. And this policy cannot use a random mac, it needs the dr/vtep mac of the other hosts by design.

Reply:
One would have hoped that the MAC can be learned through ARP. But I assume this is for added security.